find.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import os.path
+
+parser = argparse.ArgumentParser(description='find all relevant files to include in a hackpack')
+parser.add_argument('-e', '--exclude', action='append', dest='exclude', default=[], help='file to exclude')
+parser.add_argument('-f', '--first', action='append', dest='first', default=[], help='folders to place first')
+parser.add_argument('-l', '--last', action='append', dest='last', default=[], help='folders to place last')
+parser.add_argument('-g', '--first-file', action='append', dest='first_file', default=[], help='files to place first')
+parser.add_argument('-m', '--last-file', action='append', dest='last_file', default=[], help='files to place last')
+parser.add_argument('dir', nargs='?', default='.', help='scan directory')
+
+args = parser.parse_args()
+
+for idx, name in enumerate(args.exclude):
+    args.exclude[idx] = os.path.join(args.dir, name)
+
+args.first.reverse()
+args.first_file.reverse()
+
+docs = []
+
+for root, dirs, files in os.walk(args.dir):
```

Comment:
Agreed. Not quite sure what generator expression I would use and I'm not quite sure it would be more idiomatic or any faster.